File: SkyPlanner/Sales/src/Sales.Domain/Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Sales.Domain.Models
{
    public class Order: Entity<long>
    {
        public Order()
        {
            CreateDate = DateTime.Now;
            CreateDate = DateTime.Now;
        }
        
        public DateTime CreateDate { get; set; }

      
        public DateTime LastUpdateDate { get; set; }

        public float Taxes { get; set; }

        public float SubTotal { get; set; }

        public float Total => SubTotal + Taxes;

        public int AccountId { get; set; }
        public Account Account { get; set; }

        public IEnumerable<OrderProduct> OrderProducts { get; set; }

    }
}

File: SkyPlanner/Sales/src/Sales-UI/src/app/shared/core/models/order.ts
import { Entity } from './entity';
import { OrderProduct } from './order-product';
export interface Order extends Entity {
    createDate: Date;
    lastUpdateDate: Date;
    taxes: number;
    subTotal: number;
    total: number;
    orderProducts: OrderProduct[];
}
export interface AddOrder {
    taxes: number;
    subTotal: number;
    accountId: number;
    orderProducts: AddOrderProduct[];
}
export interface AddOrderProduct {
    productId: number;
    quantity: number;
}
